infra/production/test-subscription-automation-preflight-datasource.sh

```
@@ -2,264 +2,171 @@
-grep -Fxq 'DUE_CANDIDATE_COUNT=1' "$TEST_ROOT/rds-output"
-grep -Fxq 'SUBSCRIPTION_AUTOMATION_PREFLIGHT=PASS' "$TEST_ROOT/rds-output"
-[[ "$(<"$MARKER_DIR/rds-query-count")" == "2" ]]
-[[ ! -e "$MARKER_DIR/docker-query-count" ]]
-unset FAKE_FAIL_DOCKER_QUERY FAKE_FAIL_RDS_QUERY FAKE_DOCKER_DUE_COUNT FAKE_RDS_DUE_COUNT
+rm -f "$MARKER_DIR/query-count"
+if FAKE_MYSQL_MODE=failure run_preflight "$RUNTIME" "$STATE" 2 "$TEST_ROOT/failure-output"; then
+  printf 'mysql failure was reported as success\n' >&2
+  exit 1
+fi
+grep -Fq 'read-only external MySQL preflight failed; raw database output was suppressed' "$TEST_ROOT/failure-output"
+! grep -Fq 'SELECT ' "$TEST_ROOT/failure-output"
```

Shellcheck (0.11.0)

[info] This ! is not on a condition and skips errexit. Use `&& exit 1` instead, or make sure $? is checked.

(SC2251)